src/main/java/com/celements/parents/DocumentParentsLister.java

```diff
@@ -0,0 +1,79 @@
+package com.celements.parents;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.annotation.Requirement;
+import org.xwiki.model.reference.DocumentReference;
+
+import com.celements.web.service.XDocRecursionException;
+
+@Component
+public class DocumentParentsLister implements IDocumentParentsListerRole {
+
+  private static Logger _LOGGER = LoggerFactory.getLogger(DocumentParentsLister.class);
```

Comment:
Starting a variable name with underscore is not compliant with the java naming conventions, see http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-135099.html#367

"Variable names should not start with underscore _ or dollar sign $ characters, even though both are allowed."